### PR TITLE
add `inheritEnv` action parameter

### PR DESCRIPTION
**Pull request by OmarTawfik**
> Upon some further discussion and great insights from @jkeech, I think an opt-in experience (i.e. `autoEnv` or `inerhitEnv`) could make good sense. This way it doesn't change the default experience / dev container expectations for all scenarios, but it'd be a helpful option in ones such as those detailed above. And if `env` was used, it could be treated as an override to anything that was automatically inherited.

_Originally posted by @bamurtaugh in https://github.com/devcontainers/ci/issues/242#issuecomment-1642587696_

### CODE REVIEW

**Comment by OmarTawfik** [comment]
cc @bamurtaugh  and @stuartleeks

**Review by chrmarti — CHANGES_REQUESTED** [review]
Great, thanks!

Could you also add it to:
- the table in `azdo-task/README.md`
- `azdo-task/DevcontainersCi/task.json`
- the table in `docs/azure-devops-task.md`

**Comment by OmarTawfik** [comment]
@chrmarti done. Thanks.

File: common/src/envvars.ts [hunk 1]
```diff
@@ -29,8 +29,13 @@ function getSubstitutionValue(regexMatch: string, placeholder: string): string {
 // In the latter case, the corresponding returned item would be "BAR=hi"
 // where the value is taken from the matching process env var.
 // In the case of values not set in the process, they are omitted
-export function populateDefaults(envs: string[]): string[] {
+export function populateDefaults(envs: string[], inerhitEnv: boolean): string[] {
```
Comment (chrmarti) [inline common/src/envvars.ts:32]: Typo: `inerhitEnv`.

File: common/src/envvars.ts [hunk 2]
```diff
@@ -29,8 +29,13 @@ function getSubstitutionValue(regexMatch: string, placeholder: string): string {
...
 	const result: string[] = [];
+	if (inerhitEnv) {
+		for (const [key, value] of Object.entries(process.env)) {
```
Comment (chrmarti) [inline common/src/envvars.ts:35]: Should some variables be skipped? E.g., `PATH` might cause issues, maybe others.

Another approach would be to make it easy to copy over specific variables. What are the main usages for this? All CI env variables with a common prefix?

**Review by chrmarti — CHANGES_REQUESTED** [review]

File: common/src/envvars.ts (hunk 1 above)
Comment (OmarTawfik) [reply common/src/envvars.ts]: Good catch :) fixed!

File: common/src/envvars.ts (hunk 2 above)
Comment (OmarTawfik) [reply common/src/envvars.ts:35]: I'm happy to have a reject list for troublesome variables, like `PATH`, but the feature is specifically about variables from dependencies/other actions, that are not explicitly specified/controlled by the user.
Please check the discussion on #242 for more info/examples.

Comment (OmarTawfik) [reply common/src/envvars.ts:35]: Added `PATH` for now. Please let me know if I can add anything else.

**Comment by OmarTawfik** [comment]
@chrmarti done. Thanks.

**Review by chrmarti — APPROVED** [review]
Great, thanks!

**Comment by chrmarti** [comment]
CI seems stuck, closing and reopening to trigger a new run.

**Comment by chrmarti** [comment]
CI failed with:
```
      DataCloneError: [object Object] could not be cloned.
  
        53 |
        54 | 	test('inherits process env when asked', () => {
      > 55 | 		const originalEnv = structuredClone(process.env);
           | 		                    ^
        56 | 		try {
        57 | 			process.env = {TEST_ENV1: 'value1'};
        58 | 			const input = ['TEST_ENV2=value2'];
  
        at Object.<anonymous> (__tests__/envvars.test.ts:55:23)
```

**Comment by OmarTawfik** [comment]
@chrmarti fixed test by removing `structuredClone()`. The old object is preserved.

**Review by chrmarti — APPROVED** [review]

**Review by DonJayamanne — APPROVED** [review]

**Comment by OmarTawfik** [comment]
@chrmarti can you please tell me when does this get released?
Thanks!

**Comment by OmarTawfik** [comment]
Is it possible to ship a new release for this? unfortunately this repo cannot be imported/used from source. Thanks!
cc @bamurtaugh or @DonJayamanne

**Comment by Sam13** [comment]
> Is it possible to ship a new release for this? unfortunately this repo cannot be imported/used from source. Thanks! cc @bamurtaugh or @DonJayamanne

+1

**Comment by OmarTawfik** [comment]
> Is it possible to ship a new release for this? unfortunately this repo cannot be imported/used from source. Thanks! 

One more attempt? cc @bamurtaugh or @DonJayamanne or @chrmarti

**Pull request merged**